test/topology_custom/test_multidc.py

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Test functionality on the cluster with different values of the --smp parameter on the nodes.
+"""
+import logging
+import sys
+
+import pytest
+
+sys.path.insert(0, sys.path[0] + "/test/cql-pytest")
+import nodetool
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+from test.pylib.manager_client import ManagerClient
+from test.pylib.random_tables import RandomTables, TextType, Column
+from test.pylib.util import unique_name
+
+logger = logging.getLogger(__name__)
+CONFIG = {"endpoint_snitch": "GossipingPropertyFileSnitch"}
+
+
+# Checks a cluster boot/operations in multi-dc environment
+@pytest.mark.asyncio
+@pytest.mark.skip(reason="Too heavy for CI, but OK for local run")
+async def test_multidc(request: pytest.FixtureRequest, manager: ManagerClient) -> None:
+    logger.info("Creating a new node")
+    for i in range(30):
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={'dc': f'dc{i}', 'rack': 'myrack1'}
+        )
+        logger.info(s_info)
+    random_tables = RandomTables(request.node.name, manager, unique_name(), 3)
+    logger.info("Creating new tables")
+    await random_tables.add_tables(ntables=3, ncolumns=3)
+    await random_tables.verify_schema()
+
+
+@pytest.mark.asyncio
+async def test_putget_2dc_rf1(
+        request: pytest.FixtureRequest, manager: ManagerClient
+) -> None:
+    """Simple put-get test for 2 DC with one node each (RF=1) [catches #3539]"""
+    ks = "test_ks"
+    cf = "test_cf"
+    table_name = "test_table_name"
+    columns = [Column("name", TextType), Column("value", TextType)]
+    logger.info("Create two servers in different DC's")
+    for i in [1, 2]:
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={"dc": f"dc{i}", "rack": "myrack"},
+        )
+        logger.info(s_info)
+    conn = manager.get_cql()
+    random_tables = RandomTables(request.node.name, manager, ks, 1)
+    logger.info("Add table")
+    await random_tables.add_table(ncolumns=2, columns=columns, pks=1, name=table_name)
+    conn.execute(f"USE {ks}")
+    conn.execute(
+        f"CREATE COLUMNFAMILY {cf} ( key varchar, c varchar, v varchar, PRIMARY KEY(key, c)) WITH comment='test cf'"
+    )
+
+    logger.info("Perform insert/overwrite")
+    putget(conn, cf, ks)
+
+
+@pytest.mark.asyncio
+async def test_putget_2dc_rf2(
```

Comment:
Nitpick: It seems this test is identical to the previous one except for the array, so you could have done one copy of the test with "pytest.parameterize".

Another benefit is that you won't need the extra putget() function (which I, as a reader, have absolutely no idea what it does), you can insert it inside this function and let the reader read the test sequentially without jumping around the file.

Reply:
Good point. I'll rewrite it to be parametrized.